File: src/main/java/de/dk/api/codeChallenge/common/mapper/UserResponseDtoConverter.java
package de.dk.api.codeChallenge.common.mapper;

import de.dk.api.codeChallenge.notification.model.dto.UserRegisterRequestDto;
import de.dk.api.codeChallenge.notification.model.dto.UserResponseDto;
import java.util.HashSet;
import org.springframework.stereotype.Component;

@Component
public class UserResponseDtoConverter {
    public UserRegisterRequestDto convert(UserResponseDto from) {
        return new UserRegisterRequestDto(
                from.userId(),
                new HashSet<>(from.notificationTypeNames()));
    }
}

File: src/main/kotlin/de/dkb/api/codeChallenge/common/mapper/UserResponseDtoConverter.kt
package de.dkb.api.codeChallenge.common.mapper

import de.dkb.api.codeChallenge.notification.model.dto.UserRegisterRequestDto
import de.dkb.api.codeChallenge.notification.model.dto.UserResponseDto
import org.springframework.stereotype.Component


@Component
class UserResponseDtoConverter {
    fun convert(from: UserResponseDto): UserRegisterRequestDto =
        UserRegisterRequestDto(
            id = from.userId,
            notifications = from.notificationTypeNames.toMutableSet()
        )
}
